### ISSUE TITLE
Can't prevent creation of new password user without causing error message

### ISSUE THREAD

**Issue by brettle** [meteor/meteor#5110]
If a `validateLoginAttempt()` handler throws a `LoginCancelledError`, during a call to `Accounts.createUser()`, `accounts-ui-unstyled` will display the reason as an error message. By contrast, if the same error is thrown while logging in using an OAuth service, it is ignored. It should be ignored in the `Accounts.createUser()` case as well (or the reason should be displayed as an info message instead of an error). Alternatively, there should be some other way for the server to indicate that a new user wasn't created without causing an error to be displayed. `useraccounts:core` uses the `LoginCancelledError` to avoid an error message when adding an OAuth service to an existing account, and `brettle:accounts-add-service` does the same and can be used when adding a password login. It would be nice if it could use the same mechanism. 

I'm willing to create a PR that addresses this (and adds tests for it) if such a PR would be considered.

**Comment by glasser** [comment]
Thanks for the feature request! We welcome discussions about how to make Meteor better. If you haven't yet, check out our wiki page about [feature requests on GitHub](https://github.com/meteor/meteor/wiki/Contributing-to-Meteor#feature-requests).

The purpose of `Accounts.LoginCancelledError` is to indicate that the user proactively cancelled the login attempt — it's something that indicates user action, not something that a `validateLoginAttempt` callback is supposed to return from the server.

But I guess your point is that (a) anything you do in `validateLoginAttempt` to make it fail results in  *some* sort of error being returned over DDP and (b) `accounts-ui` interprets any error it gets from the network (other than LoginCancelledError which generally means the user closed the oauth popup locally) as meaning it should show an error message.

I think (a) is very reasonable.  (b)... I mean, the choice we made when writing accounts-ui is that error messages should be shown.  I'm not sure that it's a good UI choice for accounts-ui to let you try to create a user with password and for it to just not display anything on error.  You can use alternate UI packages if you want to make different choices.

I didn't follow your explanation about how it helps with these packages.  I don't understand the use case where you want an account creation attempt to fail but for the user to not be told how or why.  Can you explain in more detail?

**Comment by brettle** [comment]
Sorry for the poor explanation. Let me try again... 

The situation is a guest user (e.g. created via my `brettle:accounts-anonymous` package) adding a login service to his account. My `brettle:accounts-patch-ui` package overrides `currentUser` so that `accounts-ui-unstyled` treats guests as logged out (i.e. it lets the guest user sign-in with any of the available login services). When the guest user signs-in with one of the login services, my `brettle:accounts-add-service` package detects that the user is already logged in (as a guest in this case) and is logging in as a new user using a new service. It interprets this as a desire to add the new service to the current user's account. It adds the service to the current user's account and throws a `LoginCancelledError` from a `validateLoginAttempt()` handler. This prevents the creation of the new user and keeps the user logged in to their existing account. When the new service is an OAuth service, this works great from a UI perspective. When the new service is the password service, the error is displayed.

I understand that this isn't the original purpose of `LoginCancelledError` and am open to suggestions for better ways to achieve the desired effect. Specifically, my goals are that the user's id should not change, no new account should be created, and it should work with arbitrary login services. 

The simplest solution seems to be to expand the meaning of `LoginCancelledError` to include any cancellation by the server that can be ignored by the user (i.e. a cancellation that was initiated by the user, or a cancellation that was initiated by the server but doesn't affect the user). With that new interpretation of `LoginCancelledError`, `accounts-ui-unstyled` would need small changes to treat `LoginCancelledError` the same as success when it calls `Accounts.createUser()` [here](https://github.com/meteor/meteor/blob/devel/packages/accounts-ui-unstyled/login_buttons_dropdown.js#L432), and possibly `Meteor.loginWithPassword()` [here](https://github.com/meteor/meteor/blob/devel/packages/accounts-ui-unstyled/login_buttons_dropdown.js#L392).

**Comment by hwillson** [comment]
To help provide a more clear separation between feature requests and bugs, and to help clean up the feature request backlog, Meteor feature requests are now being managed under the https://github.com/meteor/meteor-feature-requests repository. 

This feature request will be closed here, but anyone interested in migrating this feature request to the new repository (to make sure it stays active), can click [here](https://github.com/meteor/meteor-feature-requests/issues/new?title=Can't%20prevent%20creation%20of%20new%20password%20user%20without%20causing%20error%20message&body=Migrated%20from:%20meteor/meteor%235110) to start the feature request migration process. This manual migration process is intended to help identify which of the older feature requests are still considered to be of value to the community. Thanks!
